pilot/pkg/security/trustdomain/util.go

```diff
@@ -0,0 +1,50 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package trustdomain
+
+import "strings"
+
+// stringMatch checks if a string is in a list, it supports four types of string matches:
+// 1. Exact match.
+// 2. Wild character match. "*" matches any string.
+// 3. Prefix match. For example, "book*" matches "bookstore", "bookshop", etc.
+// 4. Suffix match. For example, "*/review" matches "/bookstore/review", "/products/review", etc.
+// This is an extensive version of model.stringMatch(). The pattern can be in the string or the list.
+func stringMatch(a string, list []string) bool {
```

Comment:
This has a different behavior than the `model.stringMatch()`, we should also give it a different name to avoid confusion, also for other functions in this file.

Reply:
It's only used inside this package, so I think it's fine this way (also the comment mentions the stringMatch() from the model package)? If not, what would you name it?